code/controllers/subsystem/afk.dm

```diff
@@ -0,0 +1,81 @@
+#define AFK_WARNED       1
+#define AFK_CRYOD    	 2
+
+SUBSYSTEM_DEF(afk)
+	name = "AFK Watcher"
+	wait = 300
+	flags = SS_BACKGROUND
+	var/list/afk_players = list() // Associative list. ckey as key and AFK state as value
+
+
+/datum/controller/subsystem/afk/Initialize()
+    if(config.warn_afk_minimum <= 0 || config.auto_cryo_afk <= 0 || config.auto_despawn_afk <= 0)
+        flags |= SS_NO_FIRE
+
+/datum/controller/subsystem/afk/fire()
+	var/list/toRemove = list()
+	for(var/mob/living/carbon/human/H in GLOB.living_mob_list)
+		if(!H.ckey) // Useless non ckey creatures
+			continue
+		
+		var/turf/T 
+		// Only players and players with the AFK watch enabled
+		// No dead, unconcious, restrained, people without jobs, people on other Z levels than the station or antags
+		if(H.client == null || !H.client.prefs.afk_watch || !H.mind || \
```

Comment:
Why use H.client == null instead of !H.client?

Reply:
I think I was coding a lot in Progress that day. Where you explicitly need to say it's null. I'll change it to be more in line with the byond code